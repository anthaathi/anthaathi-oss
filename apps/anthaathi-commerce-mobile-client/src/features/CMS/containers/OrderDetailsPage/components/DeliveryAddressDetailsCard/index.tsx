import React from 'react';
import {useIntl} from 'react-intl';
import {View} from 'react-native';
import {Card, Text} from 'react-native-paper';
import {OrderDetailsPageComponentType} from '../../../../types/common';

export interface DeliveryAddressDetailsProps {
  deliveryTitle: string;
  deliveryAddress: string;
  mobileNumber: string;
}

const DeliveryAddressDetailsCard = (props: DeliveryAddressDetailsProps) => {
  const intl = useIntl();
  return (
    <Card
      style={{
        borderRadius: 4,
        margin: 10,
        padding: 15,
        borderColor: '#E3E2E7',
        borderWidth: 1,
      }}
      testID="deliveryAddressDetailsCard">
      <View>
        <Text
          style={{
            color: '#364A15',
            fontWeight: '700',
            fontSize: 16,
          }}>
          {props.deliveryTitle}
        </Text>
        <View style={{flexDirection: 'row', marginVertical: 5}}>
          <Text style={{color: '#364A15', fontSize: 14, width: '85%'}}>
            {props.deliveryAddress}
          </Text>
        </View>
        <View style={{flexDirection: 'row', marginVertical: 5}}>
          <Text style={{color: '#808080', fontSize: 14}}>
            {intl.formatMessage({defaultMessage: 'Mobile'})}
          </Text>
          <Text style={{color: '#364A15', fontSize: 14}}>
            {': ' + props.mobileNumber}
          </Text>
        </View>
      </View>
    </Card>
  );
};

export default DeliveryAddressDetailsCard;

export const DeliveryAddressDetailsCardCMSInput = {
  _component: OrderDetailsPageComponentType.DeliveryAddressDetailsCard,
  component: DeliveryAddressDetailsCard,
};
