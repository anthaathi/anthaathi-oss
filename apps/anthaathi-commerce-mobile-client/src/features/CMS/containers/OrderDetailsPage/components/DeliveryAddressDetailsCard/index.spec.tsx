import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import DeliveryAddressDetailsCard from '.';

describe('DeliveryAddressDetailsCard', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddressDetailsCard
            deliveryTitle="Delivery Address"
            deliveryAddress="USA"
            mobileNumber="+90909090"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('deliveryAddressDetailsCard')).toBeTruthy();
  });

  it('should have DeliveryAddressDetailsCard title, address and mobile number', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddressDetailsCard
            deliveryTitle="Delivery Address"
            deliveryAddress="USA"
            buttonTitle="Change"
            mobileNumber="+90909090"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Delivery Address')).toBeTruthy();
    expect(temp.queryByText('USA')).toBeTruthy();
    expect(temp.queryByText('Mobile')).toBeTruthy();
    expect(temp.queryByText(': +90909090')).toBeTruthy();
  });

});