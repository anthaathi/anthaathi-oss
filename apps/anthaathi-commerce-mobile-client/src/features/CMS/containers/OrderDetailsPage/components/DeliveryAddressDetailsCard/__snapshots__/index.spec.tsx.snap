// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryAddressDetailsCard should render the item 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#E3E2E7",
      "borderRadius": 4,
      "borderWidth": 1,
      "elevation": 1,
      "margin": 10,
      "padding": 15,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="deliveryAddressDetailsCard"
  >
    <View
      index={0}
      siblings={
        [
          "View",
        ]
      }
      total={1}
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "#364A15",
              "fontSize": 16,
              "fontWeight": "700",
            },
          ]
        }
      >
        Delivery Address
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#364A15",
                "fontSize": 14,
                "width": "85%",
              },
            ]
          }
        >
          USA
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#808080",
                "fontSize": 14,
              },
            ]
          }
        >
          Mobile
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#364A15",
                "fontSize": 14,
              },
            ]
          }
        >
          : +90909090
        </Text>
      </View>
    </View>
  </View>
</View>
`;
